packages/sdk-js/src/GrowthBook.ts

```diff
@@ -726,20 +734,33 @@ export class GrowthBook<
               id: experiment.key,
             }
           );
+          this._unsetAntiFlicker();
           return result;
         }
         this._redirectedUrl = url;
         const navigate = this._getNavigateFunction();
         if (navigate) {
           if (isBrowser) {
-            this._setAntiFlicker();
-            window.setTimeout(() => {
+            this._setAntiFlicker(true);
+            // Wait for the possibly-async tracking callback, bound by min and max delays
+            Promise.all([
+              promiseTimeout(
+                this._activeTrackingCall,
+                this._ctx.maxNavigateDelay ?? 10000
+              ),
+              new Promise((resolve) =>
+                window.setTimeout(resolve, this._ctx.navigateDelay ?? 100)
```

Comment:
Do we want this?  If the tracking call finishes fast, we shouldn't need to wait the full 100ms

Reply:
I put it here to protect against false positives (tracking complete after 0ms) / misconfigured tracking callbacks. It's adjustable to 0. We can remove it if you prefer though, but it does seem useful for things like the script tag